pkg/apis/core/types.go

```diff
@@ -425,25 +425,53 @@ type PersistentVolumeClaimList struct {
 // PersistentVolumeClaimSpec describes the common attributes of storage devices
 // and allows a Source for provider-specific attributes
 type PersistentVolumeClaimSpec struct {
-	// Contains the types of access modes required
+	// Contains the types of access modes the volume should have.
+	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes-1
 	// +optional
 	AccessModes []PersistentVolumeAccessMode
 	// A label query over volumes to consider for binding. This selector is
-	// ignored when VolumeName is set
+	// ignored when volumeName is set.
+	// A claim with a non-empty selector can't have a PersistentVolume
+	// dynamically provisioned for it.
 	// +optional
 	Selector *metav1.LabelSelector
-	// Resources represents the minimum resources required
+	// Represents the minimum resources the volume should have.
+	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#resources
 	// +optional
 	Resources ResourceRequirements
-	// VolumeName is the binding reference to the PersistentVolume backing this
-	// claim. When set to non-empty value Selector is not evaluated
+	// The binding reference to the PersistentVolume backing this claim. When
+	// set to non-empty value, selector is not evaluated.
 	// +optional
 	VolumeName string
-	// Name of the StorageClass required by the claim.
-	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1
+	// Name of the StorageClass required by the claim. Only PersistentVolumes of
+	// the requested class can be bound to the claim.
+	// If the value is "" (an empty string), the claim can only be bound to a
+	// PersistentVolume with no StorageClass (the storageClassName field is
+	// either unset or set to "").
+	// If this field is not set, the behavior depends on whether the
```

Comment:
This might be a bit too much for the API reference. I think we can reduce this down to default storageclass if the admission controller is running vs not. But leave the details about how to set the default storageclass to the docs.